### PR TITLE
Views 

### PR DESCRIPTION
**Pull request by niicode**
In this PR, I added:
Logic to the post index and show
Logic to the users index and show
Partials for the users and view template

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,8 @@
-<h2>Home page</h2>
+<h1> List all users </h1>
+<div>
+   <ul class='list'>
+      <% @users.each do |user| %>
+         <%= render 'shared/user', user: user %>
```
Comment (fobadara) [inline app/views/users/index.html.erb]: [OPTIONAL] I think you should add a `li` tag here since. you have a `ul` wrapping it

**Review by fobadara — CHANGES_REQUESTED** [review]
# REQUIRES CHANGES ♻️ 
Hi @niicode, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Linters are passing
✅ Project looks professional


## Required Changes ♻️
_Check the comments under the review._
I had to include. some comments here because they could not be done inline.
[REQUIRED] When I run your migrations I get errors:
image
Kindly fix this error for further checks from the next reviewer

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1 +1,13 @@
...
+    <h3> <%=@post.title%> by <%= @user.name%> </h3>
+    <p> <%= @post.text%> </p>
+    <p> Comments: <%= @post.comments_counter %>, Likes: <%= @post.likes_counter %> </p>
+  </div>
+  <div>
+    <% @post.recent_five_comments.each do |comment| %>
+      <p> <%= comment.user.name%>: <%= comment.text%><p>
+    <% end %>
+  </div>
+</div>
```
Comment (DammyShittu) [inline app/views/posts/show.html.erb]: -[ ] Kindly style your post show page to match the wireframe: [https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=GHSAT0AAAAAABY4RVENF5O6IYX2BVYE7R7GY4J3RNQ](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=GHSAT0AAAAAABY4RVENF5O6IYX2BVYE7R7GY4J3RNQ)

File: app/views/posts/show.html.erb [hunk 3]
```diff
@@ -1 +1,13 @@
-<h2>This is the show page for posts</h2>
+<h1> show post for a given user</h1>
+<div>
+  <div>
+    <h3> <%=@post.title%> by <%= @user.name%> </h3>
+    <p> <%= @post.text%> </p>
+    <p> Comments: <%= @post.comments_counter %>, Likes: <%= @post.likes_counter %> </p>
+  </div>
+  <div>
+    <% @post.recent_five_comments.each do |comment| %>
```
Comment (DammyShittu) [inline app/views/posts/show.html.erb]: - [x] I get this error when I visit the post show page. Kindly fix that. What you have is `last_five_comments`, not `recent_five_comments`.

  didow

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1 +1,19 @@
...
+  <%= render 'shared/user', user: @user %>
+
+  <div class='bio'>
+    <p><%= @user.bio %></p>
+  </div>
+
+  <div class='card'>
+      <% @user.three_recent_posts.each do |post| %>
+        <a href="/users/<%= @user.id %>/posts/<%= post.id %> ">
+          <%= render 'shared/post', post: post %>
```
Comment (DammyShittu) [inline app/views/users/show.html.erb]: - [ ] Kindly style the post part for the user show page to match the wireframe: [https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_page.png?token=GHSAT0AAAAAABY4RVENCWGONH3SFHF5DZB4Y4J3SSA](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_page.png?token=GHSAT0AAAAAABY4RVENCWGONH3SFHF5DZB4Y4J3SSA)

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @niicode,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @niicode, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Views ✔
- No linter errors ✔
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**